consul/structs/prepared_query.go

```diff
@@ -72,6 +72,16 @@ type PreparedQuery struct {
 	RaftIndex
 }
 
+// GetACLPrefix returns the prefix to look up the prepared_query ACL policy for
+// this query, or nil if such a policy doesn't apply.
+func (pq *PreparedQuery) GetACLPrefix() *string {
```

Comment:
I feel like this should return like a `(string, bool)` tuple, and then use the `if prefix, ok := ...` pattern. Any other reason to use a pointer?